code/controllers/subsystem/economy.dm

```diff
@@ -48,46 +75,78 @@ SUBSYSTEM_DEF(economy)
 
+/datum/controller/subsystem/economy/proc/get_bank_account_by_id(target_id)
+	if(!length(bank_accounts))
+		return FALSE
+	if(istype(target_id, /datum/bank_account))
+		stack_trace("proc took account type itself, but it is supposed to take account id number.")
+		return target_id
+	target_id = text2num(target_id)
+	for(var/datum/bank_account/target_account in bank_accounts)
+		if(target_account.account_id == target_id)
+			return target_account
+	return null
+
 /datum/controller/subsystem/economy/proc/get_dep_account(dep_id)
+	if(united_budget_trait && !(dep_id in nonstation_accounts)) // using 'is_nonstation()' here is innecessary
+		dep_id = ACCOUNT_CAR_ID
 	for(var/datum/bank_account/department/D in generated_accounts)
 		if(D.department_id == dep_id)
 			return D
 
+/datum/controller/subsystem/economy/proc/get_dept_id_by_bitflag(target_bitflag)
+	for(var/B in account_bitflags)
+		if(account_bitflags[B] == target_bitflag)
+			return B
+
+/// How to use: `bankaccount.is_nonstation_account()`, `SSeconomy.is_nonstation_account(bankaccount type)` or `SSecononmy.is_nonstation_account(ACCOUNT_CAR)`. This returns TRUE if an account is non-station account - i.e.) VIP budget, Command budget
+/datum/controller/subsystem/economy/proc/is_nonstation_account(datum/bank_account/department/D) // takes a bank account type or dep_ID define
+	if(!D) // null check first
+		return FALSE
+	if(!istype(D, /datum/bank_account/department)) // if parameter was given as a dept id, replace it into better type
+		D = SSeconomy.get_dep_account(D) // tricky
+	if(!istype(D, /datum/bank_account/department)) // if it failed to replacing, return false.
+		return FALSE
+	for(var/each in SSeconomy.nonstation_accounts)
+		if(D.account_holder == SSeconomy.nonstation_accounts[each])
+			return TRUE
+	return FALSE
+
 /datum/controller/subsystem/economy/proc/distribute_funds(amount)
-	var/datum/bank_account/eng = get_dep_account(ACCOUNT_ENG)
-	var/datum/bank_account/sec = get_dep_account(ACCOUNT_SEC)
-	var/datum/bank_account/med = get_dep_account(ACCOUNT_MED)
-	var/datum/bank_account/srv = get_dep_account(ACCOUNT_SRV)
-	var/datum/bank_account/sci = get_dep_account(ACCOUNT_SCI)
-	var/datum/bank_account/civ = get_dep_account(ACCOUNT_CIV)
-	var/datum/bank_account/car = get_dep_account(ACCOUNT_CAR)
+	var/datum/bank_account/eng = get_dep_account(ACCOUNT_ENG_ID)
+	var/datum/bank_account/sec = get_dep_account(ACCOUNT_SEC_ID)
+	var/datum/bank_account/med = get_dep_account(ACCOUNT_MED_ID)
+	var/datum/bank_account/srv = get_dep_account(ACCOUNT_SRV_ID)
+	var/datum/bank_account/sci = get_dep_account(ACCOUNT_SCI_ID)
+	var/datum/bank_account/civ = get_dep_account(ACCOUNT_CIV_ID)
+	var/datum/bank_account/car = get_dep_account(ACCOUNT_CAR_ID)
 
 	var/departments = 0
 
 	if(eng)
-		departments += 2
+		departments += BUDGET_RATIO_TYPE_B
 	if(sec)
-		departments += 2
+		departments += BUDGET_RATIO_TYPE_B
 	if(med)
-		departments += 2
+		departments += BUDGET_RATIO_TYPE_B
```

Comment:
rename to double/single.

also maybe make each department have its own `datum/bank_account` subtype with a ratio stored onto it instead of this awful if statement